editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/preferences/PreferredLabelEditor.java

```diff
@@ -0,0 +1,424 @@
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.preferences;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.preference.FieldEditor;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.DisposeEvent;
+import org.eclipse.swt.events.DisposeListener;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.widgets.TableColumn;
+import org.eclipse.swt.widgets.TableItem;
+import org.eclipse.swt.widgets.Widget;
+import org.fusesource.ide.camel.editor.internal.UIMessages;
+import org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement;
+
+/**
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class PreferredLabelEditor extends FieldEditor {
+
+	private Table table;
+
+	/**
+	 * The button box containing the Add, Edit and Remove buttons.
+	 */
+	private Composite buttonBox;
+
+	private Button addButton;
+	private Button editButton;
+	private Button removeButton;
+
+	/**
+	 * The selection listener.
+	 */
+	private SelectionListener selectionListener;
+
+	/**
+	 * Creates a new list field editor
+	 */
+	public PreferredLabelEditor() {
+	}
+
+	/**
+	 * Creates a table field editor.
+	 *
+	 * @param name
+	 *            the name of the preference this field editor works on
+	 * @param labelText
+	 *            the label text of the field editor
+	 * @param parent
+	 *            the parent of the field editor's control
+	 */
+	public PreferredLabelEditor(String name, String labelText, Composite parent) {
+		init(name, labelText);
+		createControl(parent);
+	}
+
+	@Override
+	protected void adjustForNumColumns(int numColumns) {
+		Control control = getLabelControl();
+		((GridData) control.getLayoutData()).horizontalSpan = numColumns;
+		((GridData) table.getLayoutData()).horizontalSpan = numColumns - 1;
+
+	}
+
+	@Override
+	protected void doFillIntoGrid(Composite parent, int numColumns) {
+		Control control = getLabelControl(parent);
+		GridData gd = new GridData();
+		gd.horizontalSpan = numColumns;
+		control.setLayoutData(gd);
+
+		table = getTableControl(parent);
+		gd = new GridData(GridData.FILL_HORIZONTAL);
```

Comment:
you can use also GridDataFactory.fillDefaults().span(numColumns -1, 1).grab(true, false).create()

(personally I prefer, and while writing it it is faster, but not sure it is worthy to change it in existing code)